### PR TITLE
chore: various small changes

### PR DESCRIPTION
**Pull request by vvo**
- eslint 1.0
- results => hits widget
- more documentation on widgets
- readme TOC
- various indenting fixes, trying to follow react style
- consistent requires
- missing use strict

I will try to configure eslint to check the various linting changes so that it's automatically checked

### CODE REVIEW

**Comment by vvo** [comment]
Do not merge PR, I will have to reopen it to develop.

But you can still review.

File: components/SearchBox/index.js [hunk 1]
```diff
@@ -19,14 +17,14 @@ var SearchBox = React.createClass({
...
-      className={classNames}
-      data-role="autocomplete"
-      autoComplete="off"
-      autoFocus="autofocus"
-      onChange={this.change}
-      role="textbox" />
+        placeholder={this.props.placeholder}
+        name="algolia-query"
+        className={classNames}
+        data-role="autocomplete"
```
Comment (bobylito) [inline components/SearchBox/index.js]: I think this data-role is probably useless. I guess there should be a `role="textbox"` and we should use `aria-autocomplete="list"`

Comment (bobylito) [inline components/SearchBox/index.js]: https://msdn.microsoft.com/en-us/library/windows/apps/hh968002.aspx

**Comment by bobylito** [comment]
:+1:

File: components/SearchBox/index.js (hunk 1 above)
Comment (vvo) [inline components/SearchBox/index.js]: I would add this only for autocomplete.js where the suggestions are only dependent of the search input. Whereas in a an instant ui the suggestion are not dedicated to the search input but to all the interface and filters. We need an accessibility expert that will review theses questions for days :D

**Pull request merged**